doc/ContribGuide/testing.rst

```diff
@@ -0,0 +1,70 @@
+
+Before opening a PR, a minimum set of tests should be run: 
+
+   * Developers must run the fundamental test suite manually on at least one supported platform and report on the outcome in the PR template. Developers should test code modifications on as many platforms as possible. 
+
+      * To run the fundamental tests manually, run the following command from the ``tests/WE2E`` directory:
+
+        .. code-block:: console
+
+           ./run_WE2E_tests.py -t=fundamental -m=your_machine -a=your_account
+
+        where ``your_machine`` is the Tier-1 machine you are running the tests on, and ``your_account`` is the account you charge your computational resources to. Refer to the :ref:`WE2E Tests <WE2E_tests>` chapter of the User's Guide for more detail on how to run SRW App tests. 
+
+      * Developers will not be required to run tests on *all* supported platforms, but if a failure is pointed out by another reviewer (or by automated testing), then it is expected that the developer will work with reviewers and code managers to ensure that the problem is resolved prior to merging. 
+
+   * If the PR impacts functionality contained within comprehensive WE2E tests not included in the fundamental test suite, the developer must run those tests on the PR. 
+   * Any new functionality must be tested explicitly, and any new tests should be described in detail in the PR message. Depending on the impact of this functionality, new tests should be added to the suite of comprehensive WE2E tests, followed by a discussion with code managers on whether they should also be included as fundamental tests.
+
+      * In some cases, it may be possible to modify a current test instead of creating a completely new test. Code developers introducing new capabilities should work with code managers to provide the proper configuration files, data, and other information necessary to create new tests for these capabilities.
+
+   * When the above tests are complete and the PR has been approved by at least one code manager, a code manager will add the ``run_we2e_coverage_tests`` label to initiate fundamental testing on all Level 1 platforms via the Jenkins CI/CD pipeline.
+
+Testing on Jenkins
+===================
+
+`Jenkins <https://www.jenkins.io/>`__ is an "open source automation server" that automates code testing. For the Jenkins automated testing labels, it should be noted that **ONLY** code managers should apply these labels and only after at least one code manager has given approval to the PR.  The PR will not be merged until all Jenkins-based builds and testing have successfully passed.
+
+The following automated testing labels are available for the SRW App:
+
+   * ``run_we2e_coverage_tests``
+   * ``run_we2e_comprehensive_tests``
```

Comment:
Since the `run_we2e_comprehensive_tests` label isn't currently used, should we remove it until such time that it is?

Reply:
I changed it to _Coming Soon_, as it is on the wiki. Do you prefer for me to remove it entirely?

Reply:
Adding  _Coming Soon_ is fine!  Thanks for making the change!